app/models/assignment.rb

```diff
@@ -1247,29 +1247,122 @@ def to_json(options = {})
+    elsif self.is_peer_review?
+      peer_fields = [:group_min, :group_max, :student_form_groups, :assign_graders_to_criteria,
+                     :group_name_autogenerated, :group_name_displayed, :invalid_override, :section_groups_only,
+                     :section_due_dates_type, :anonymize_groups, :hide_unassigned_criteria]
+      peer_fields.each { |f| other_properties[f] = self.send(f) }
+      properties.merge('assignment_properties_attributes': other_properties)
+    else
+      assignment_fields = [:group_min, :group_max, :allow_web_submits, :student_form_groups, :remark_due_date,
+                           :remark_message, :assign_graders_to_criteria, :allow_remarks,
+                           :display_grader_names_to_students, :display_median_to_students, :group_name_autogenerated,
+                           :vcs_submit, :has_peer_review, :group_name_displayed, :invalid_override,
+                           :section_groups_only, :section_due_dates_type, :only_required_files, :anonymize_groups,
+                           :hide_unassigned_criteria, :duration, :start_time, :is_timed]
+      assignment_fields.each do |f|
+        other_properties[f] = self.send(f) if self.is_timed || [:is_timed, :start_time, :duration].exclude?(f)
+      end
+      # Get all other assignment properties
+      late_periods = self.submission_rule.periods.pluck_to_hash(:deduction, :hours, :interval)
+      required_files = self.assignment_files.pluck_to_hash(:filename)
+      # Merge all properties together
+      properties.merge('assignment_properties_attributes': other_properties,
+                       'submission_rule_attributes': { 'type': self.submission_rule.type,
+                                                       'periods_attributes': late_periods },
+                       'assignment_files_attributes': required_files)
+    end
+  end
+
+  # Writes this assignment's starter file settings to the file located at +settings_filepath+ located in the
+  # +zip_file+. Also writes the starter files for this assignment in the same directory as +settings_filepath+.
+  def starter_file_config_to_zip(zip_file, settings_filepath)
+    default_starter_group = nil
+    group_data = []
+    directory_path = File.dirname(settings_filepath)
+    self.starter_file_groups.each.with_index(1) do |starter_file_group, download_count|
+      group_name = "#{download_count}: #{starter_file_group.name}"
+      group_name = group_name.gsub(File::SEPARATOR, '')
+      starter_file_group.write_starter_files_to_zip(zip_file, File.join(directory_path, group_name))
+      if starter_file_group.id == self.default_starter_file_group_id
+        default_starter_group = group_name
+      end
+      group_data << {
+        directory_name: group_name,
+        name: starter_file_group.name,
+        use_rename: starter_file_group.use_rename,
+        entry_rename: starter_file_group.entry_rename
+      }
+    end
+    starter_file_settings = {
```

Comment:
For the same reason as in the comment for `assignment_properties_config` I would like to avoid explicitly including attribute names whenever possible. I know that you want to keep all the information about starter files together in one zip file entry but in this case, the `starter_file_type` and the `starter_files_after_due` attributes should really be with the assignment properties, not here.

If we don't set them initially when we create the assignment then we run the risk of the new assignment not being valid (for example, if one of these attributes becomes a required attributes later on)